### PR TITLE
PC-942: Update dependencies

### PR DESCRIPTION
**Pull request by ulfryk**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,39 +31,39 @@
     "node": ">=6"
   },
   "dependencies": {
-    "@samwise-tech/core": "0.0.3",
+    "@samwise-tech/core": "^0.1.0",
```
Comment (dumpstate) [inline package.json]: Though.. I'd consider not using `^` for our dependencies.

Comment (ulfryk) [reply package.json]: we better do this in libs that can be dependecies of apps to avoid bundling several versions of some apps' dependencies ;)

Comment (dumpstate) [reply package.json]: imo this way we loose control about the version of this dependencies.

e.g.

`cv-webapp` <- `search-ql` <- `core`

Assuming, `cv-webapp` doesn't depend on `core` only `yarn` keeps track of the version of `core`.

File: package.json [hunk 2]
```diff
@@ -31,39 +31,39 @@
...
     "parsimmon": "1.6.2"
   },
   "devDependencies": {
-    "@samwise-tech/immutable": "^0.0.1",
-    "@samwise-tech/tslint-config": "0.4.0",
+    "@samwise-tech/immutable": "^0.0.2",
+    "@samwise-tech/tslint-config": "0.4.1",
     "@types/chai": "4.0.4",
-    "@types/lodash": "4.14.74",
+    "@types/lodash": "4.14.78",
```
Comment (tomecko) [inline package.json]: `^4.14.74`? that's what is specified in some other repos

File: package.json (hunk 1 above)
Comment (ulfryk) [reply package.json]: Few other libs depend on core, it's better to have the newest common version of it…

File: package.json (hunk 2 above)
Comment (ulfryk) [reply package.json]: It's only in `dev` dependencies (not in `dependencies` nor in `peerDependencies`) so `^` is not needed here..

File: package.json (hunk 1 above)
Comment (dumpstate) [reply package.json]: I think you meant `it's easier to have the newest common version of it` ;)

**Pull request merged**